client/tokenized-payment-request/payment-request.js

```diff
@@ -195,30 +198,35 @@ export default class WooPaymentsPaymentRequest {
 				return;
 			}
 
-			// First check if product can be added to cart.
-			if ( $addToCartButton.is( '.disabled' ) ) {
-				event.preventDefault(); // Prevent showing payment request modal.
-				if ( $addToCartButton.is( '.wc-variation-is-unavailable' ) ) {
-					window.alert(
-						window.wc_add_to_cart_variation_params
-							?.i18n_unavailable_text ||
+			// If the button is on a product page, we need to add the product to an anonymous cart before proceeding.
+			if ( getPaymentRequestData( 'button_context' ) === 'product' ) {
+				// First check if product can be added to cart.
+				if ( $addToCartButton.is( '.disabled' ) ) {
+					event.preventDefault(); // Prevent showing payment request modal.
+					if (
+						$addToCartButton.is( '.wc-variation-is-unavailable' )
+					) {
+						window.alert(
+							window.wc_add_to_cart_variation_params
+								?.i18n_unavailable_text ||
+								__(
+									'Sorry, this product is unavailable. Please choose a different combination.',
+									'woocommerce-payments'
+								)
+						);
+					} else {
+						window.alert(
 							__(
-								'Sorry, this product is unavailable. Please choose a different combination.',
+								'Please select your product options before proceeding.',
```

Comment:
I'm not necessarily sure why we need to lean on WC Core's default messaging here, but [there is another `i18n_make_a_selection_text` option](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-frontend-scripts.php#L600) that we could include here. 🤷 

Something like the below...
```js
						window.alert(
							window?.wc_add_to_cart_variation_params
								?.i18n_make_a_selection_text ||
								__(
									'Please select some product options before adding this product to your cart.',
									'woocommerce-payments'
								)
						);
```

Reply:
I'm also not sure. I think this was copied over from the [original PRB implementation](https://github.com/Automattic/woocommerce-payments/blob/04b046546d34245a85fa2b29b1567b2cc25d72ac/client/payment-request/index.js#L434-L441) in @frosso tokenized cart PDP implementation and is now showing up as a commit here. What I mean is, I didn't make this change in this PR and this is [already in `develop`](https://github.com/Automattic/woocommerce-payments/blob/04b046546d34245a85fa2b29b1567b2cc25d72ac/client/tokenized-payment-request/payment-request.js#L210-L217) from the PDP tokenized cart PR. That said, may as well go ahead and update it. Thanks for pointing it out! - [c6ab890](https://github.com/Automattic/woocommerce-payments/pull/8891/commits/c6ab890ac4ccdcd74b70569424e0696db832edd0)